src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -49,7 +50,15 @@ class MariaDBManager implements IQueryService, INotificationService, SampleEvent
 
   @Inject
   MariaDBManager(QBiCDataSource dataSource) {
-    this.dataSource = dataSource.getSource()
+    this.dataSource = dataSource
+  }
+
+  private Connection getConnection() throws UnrecoverableException {
+    try {
+      return dataSource.getConnection()
+    } catch (TimeOutException exception) {
+      throw new UnrecoverableException("Connection to data source timed out", exception)
```

Comment:
Where do we catch that exception? You mentioned that it might be a possibility to exit the application. Did you reject this idea?

Reply:
There is a global exception handler returning 500 and logging the exception

Reply:
@KochTobi exactly

Reply:
@jenniferboedker i can query the server health api and also make settings via watchdog in theory

Reply:
this solution will now return the correct HTTP response code